src/core-asyncpipeline/AsyncPipeline/AsyncResultFlow/Fold/Fold.Task.cs

```diff
@@ -0,0 +1,44 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncResultFlow<TSuccess, TFailure>
+    {
+        public AsyncPipeline<TResult> Fold<TResult>(
```

Comment:
I suppose _async_ `Fold/FoldValue` funcs should have the only overload with both _async_ `mapSuccessAsync/mapFailureAsync` strategies.

The _Fold_ func can be treated as mapping of input values into a result of the same type (that doesn't look forwarding, etc. - the result is already received, but just in tagged representation)
The success and failure transforming look equal for mapping case.

So, let's reduce the overloads.

Whether a user needs for, e.g., _async_ success mapping and _sync_ failure mapping in the same _Fold_, they may use `[Value]Task.FromResult` for the the _sync_ strategy, or consider to add a _Forward_ step before the _Fold_.

Reply:
Done